airbyte-cdk/python/airbyte_cdk/sources/declarative/schema/json_file_schema_loader.py

```diff
@@ -89,3 +95,38 @@ def extract_resource_and_schema_path(json_schema_path: str) -> (str, str):
             return "", split_path[0]
 
         return split_path[0], "/".join(split_path[1:])
+
+    def __resolve_schema_references(self, raw_schema: dict) -> dict:
+        """
+        Resolve links to external references and move it to local "definitions" map.
+
+        :param raw_schema jsonschema to lookup for external links.
+        :return JSON serializable object with references without external dependencies.
+        """
+
+        package = importlib.import_module(self.__resource)
+        base = os.path.dirname(package.__file__) + "/"
+        resolved = jsonref.JsonRef.replace_refs(raw_schema, loader=JsonFileLoader(base, "schemas/shared"), base_uri=base)
+        resolved = resolve_ref_links(resolved)
+        return resolved
+
+
+def resolve_ref_links(obj: Any) -> Union[Dict[str, Any], List[Any]]:
```

Comment:
Can you add some unit tests to validate some each of the scenarios that can be recursed.